### ISSUE TITLE
Storage is weird

### ISSUE THREAD

**Issue by CSdiscoveries** [DonBruce64/MinecraftTransportSimulator#1815]
Storage for crates (and any other similar things, such as truck beds or shipping containers), regardless of which pack they are from, are difficult to use. To summarize a long list of complaints; it's simply not the vanilla storage system. Nothing can be swapped around or sorted once already in a storage item, and mods that allow for easier item transference (for example, "Mouse Tweaks") are incompatible. Overall, the system feels clunky and difficult to use. And it doesn't seem like there should be any reason for storage to behave in this way, given how mules and such are already entities with storage (as well as the fact that shulker boxes exist).

**Comment by DonBruce64** [comment]
All of those systems you reference use MC-specific code that only works on those entities.  You cannot make a "chest" in the world anywhere.  It has to be tied to a block.  Same with mule inventories (they are tied only to mules).  I have had to roll my own code to allow for inventories on other systems.  Granted, there are some things that don't interface as tightly as other things would since they aren't typical "containers", but the reason for this is to allow more basic GUIs.

In a nutshell however, vehicle inventories are NOT designed to be sortable, long-term storage systems.  They're designed to hold ammo, fuel, and bulk transport goods.   And because of that they come with some limitations.

I'm open to folks taking a second look at the code, but I won't change them since they've always been a PITA come version update day since MC and Forge both change their inventory code every version.
